File: CS/DXWebApplication1/Services/CustomReportDesignerExceptionHandler.cs
using System;
using System.IO;
using System.ServiceModel;
using DevExpress.XtraReports.Web.ReportDesigner.Services;

namespace DXWebApplication1.Services {
    public class CustomReportDesignerExceptionHandler : ReportDesignerExceptionHandler {
        public override string GetUnknownExceptionMessage(Exception ex) {
            if (ex is FileNotFoundException) {
#if DEBUG
                return $"Debug mode. {ex.Message}.";
#else
                return "File is not found.";
#endif
            }

            return $"{ex.GetType().Name} occured. See the log file for more details.";
        }

        public override string GetFaultExceptionMessage(FaultException ex) {
            return $"FaultException occurred: {ex.Message}.";
        }
    }
}

File: VB/DXWebApplication1/Services/CustomReportDesignerExceptionHandler.vb
Imports System
Imports System.IO
Imports System.ServiceModel
Imports DevExpress.XtraReports.Web.ReportDesigner.Services

Namespace DXWebApplication1.Services
	Public Class CustomReportDesignerExceptionHandler
		Inherits ReportDesignerExceptionHandler

		Public Overrides Function GetUnknownExceptionMessage(ByVal ex As Exception) As String
			If TypeOf ex Is FileNotFoundException Then
#If DEBUG Then
				Return "Debug mode. " & ex.Message & "."
#Else
				Return "File is not found."
#End If
			End If

			Return ex.GetType().Name & " occurred. See the log file for more details."
		End Function

		Public Overrides Function GetFaultExceptionMessage(ByVal ex As FaultException) As String
			Return "FaultException occurred: " & ex.Message & "."
		End Function
	End Class
End Namespace